test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1723,6 +1723,120 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+      clientEncryption = new this.configuration.mongodbClientEncryption.ClientEncryption(client, {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: getKmsProviders()
+      });
+
+      // Using client_encryption, create a data key with a local KMS provider and the keyAltName "def".
+      setupKeyId = await clientEncryption.createDataKey('local', {
+        keyAltNames: ['def']
+      });
+    });
+
+    afterEach(async () => {
+      clientEncryption = null;
+      setupKeyId = null;
+      await client.close();
+    });
+
+    context('Case 1', metadata, function () {
+      it('createDataKey() handles duplicate key errors on the keyvault collection', async function () {
+        // 1. Use client_encryption to create a new local data key with a keyAltName "abc" and assert the operation does not fail.
+        await clientEncryption.createDataKey('local', {
+          keyAltNames: ['abc']
+        });
+
+        // 2. Repeat Step 1 and assert the operation fails due to a duplicate key server error (error code 11000).
+        const maybeErrorStep2 = await clientEncryption
+          .createDataKey('local', {
+            keyAltNames: ['abc']
+          })
+          .catch(e => e);
+        expect(
+          maybeErrorStep2,
+          'Error in step 2) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+        ).to.be.instanceof(MongoServerError);
+        expect(maybeErrorStep2).have.property('code', 11000);
+
+        // 3. Use client_encryption to create a new local data key with a keyAltName "def" and assert the operation fails due to a duplicate key server error (error code 11000).
+        const maybeErrorStep3 = await clientEncryption
+          .createDataKey('local', {
+            keyAltNames: ['def']
+          })
+          .catch(e => e);
+        expect(
+          maybeErrorStep3,
+          'Error in step 3) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+        ).to.be.instanceof(MongoServerError);
+        expect(maybeErrorStep3).have.property('code', 11000);
+      });
+    });
+
+    context('Case 2', metadata, function () {
+      it('addKeyAltName() handles duplicate key errors on the keyvault collection', async function () {
+        // 1. Use client_encryption to create a new local data key and assert the operation does not fail.
+        const _id = await clientEncryption.createDataKey('local');
+
+        // 2. Use client_encryption to add a keyAltName "abc" to the key created in Step 1 and assert the operation does not fail.
+        await clientEncryption.addKeyAltName(_id, 'abc');
+
+        // 3. Repeat Step 2, assert the operation does not fail, and assert the returned key document contains the keyAltName "abc" added in Step 2.
+        const maybeErrorStep3 = await clientEncryption.addKeyAltName(_id, 'abc').catch(e => e);
```

Comment:
I think in the cases where we expect success (here and in step 5), we don't need the catch and assign, since if it's an error, it'll get thrown and the test will fail anyway (like we have in the _id step above); if we're worried that an error is returned but not thrown from the method, we can still do the expect not error check on the result